aidb/engine/engine.py

```diff
@@ -14,7 +14,13 @@ def execute(self, query: str, **kwargs):
     try:
       parsed_query = Query(query, self._config)
       all_queries = parsed_query.all_queries_in_expressions
+
+      # FIXME: We have many validity checks for different queries.
+      #     It's better to put them together and check the validity first.
+      # check validity of user defined function query
+      _ = parsed_query.is_udf_query_valid
```

Comment:
Does this throw? Imo this should probably return True/False and the caller should throw?

Reply:
Could you please check the new commit? I have separated it into two functions: 'is_udf_query' and 'check_udf_query_validity'. As mentioned in issue #157, we could consider merging the 'check_validity' function later.